packages/intelligence/context.py

```
@@ -0,0 +1,205 @@
+        if not old_msgs:
+            return window
+
+        # Create a simple summary (full LLM summarization will be added)
+        summary_parts = []
+        for msg in old_msgs:
+            content = msg.get("content", "")
+            summary_parts.append(f"[{msg.get('role', '?')}]: {content[:100]}...")
+
+        summary_msg = {
+            "role": "system",
+            "content": f"Previous conversation summary:\n" + "\n".join(summary_parts),
```

Ruff (0.15.20)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)